route_generator/src/route_generator.cpp

```diff
@@ -74,6 +76,90 @@ bool RouteGenerator::set_active_route_cb(cav_srvs::SetActiveRouteRequest &req, c
     return true;
 }
 
+bool RouteGenerator::set_active_route_cb_new(cav_srvs::SetActiveRouteRequest &req, cav_srvs::SetActiveRouteResponse &resp)
+{
+    // load destination points
+    std::string route_file_name = route_file_path_ + req.routeID.append(".csv");
+    std::ifstream fs(route_file_name);
+    std::string line;
+    std::vector<tf2::Vector3> destination_points;
+    while(std::getline(fs, line))
+    {
+        wgs84_utils::wgs84_coordinate coordinate;
+        auto comma = line.find(",");
+        coordinate.lon = std::stod(line.substr(0, comma));
+        line.erase(0, comma + 1);
+        comma = line.find(",");
+        coordinate.lat = std::stod(line.substr(0, comma));
+        coordinate.elevation = std::stod(line.substr(comma + 1));
+        destination_points.emplace_back(wgs84_utils::geodesic_to_ecef(coordinate, tf2::Transform()));
+    }
+    // transform to local map frame
+    tf2_ros::Buffer tfBuffer;
+    tf2_ros::TransformListener tfListener(tfBuffer);
+    tf2::Transform map_in_earth;
+    try{
+        // TODO This line gives me an error.
+        //tf2::convert(tfBuffer_.lookupTransform("earth", "map", ros::Time(0)).transform, map_in_earth);
+    }
+    catch (tf2::TransformException &ex) {
+        ROS_ERROR_STREAM("Could not lookup transform with exception " << ex.what());
+    }
+    // Compose routing function inputs
+    std::vector<lanelet::BasicPoint2d> destination_points_in_map;
+    for(tf2::Vector3 point : destination_points) {
+        tf2::Transform point_in_earth;
+        point_in_earth.setOrigin(point);
+        auto point_in_map = map_in_earth.inverse() * point_in_earth;
+        destination_points_in_map.push_back(lanelet::BasicPoint2d(point_in_map.getOrigin().getX(), point_in_map.getOrigin().getY()));
+    }
+    carma_wm::LaneletRoutingGraphConstPtr p = wm->getMapRoutingGraph();
+    if(destination_points_in_map.size() < 2) {
+        return false;
+    }
+    // generate a route
+    auto route = routing(destination_points_in_map.front(),
+                         std::vector<lanelet::BasicPoint2d>(destination_points_in_map.begin() + 1, destination_points_in_map.end() - 1),
+                         destination_points_in_map.back(),
+                         wm->getMap(), wm->getMapRoutingGraph());
+    if(!route) {
+        return false;
```

Comment:
Instead of returning false we should have the service response return an indication of why the failure occurred. In this case the "No path found to destination"